lib/queue/events.ts

```diff
@@ -260,34 +131,32 @@ export const wait = async (
 		stream.once('closed', () => {
 			stream.removeAllListeners();
 			logger.info(logContext, 'Closing wait stream', {
-				slug,
+				id: actionRequestId,
 			});
 
 			resolve(result);
 		});
 
-		// Don't perform a "get by slug" if we already have a match.
+		// Don't attempt a direct query if we already have a match.
 		if (result) {
```

Comment:
I think it's impossible for this to be truthy since `stream.once('data', (change) => {` has its callback called asynchronously whilst this check is done synchronously and therefore there's never been a chance for the stream to find a result

Reply:
I agree, this is probably impossible to be truthy. See my other comment, but I'm also pretty sure we should query first and then open a stream if no matches are immediately found.